physical/raft/raft.go

```diff
@@ -652,23 +624,151 @@ func (b *RaftBackend) DisableUpgradeMigration() (bool, bool) {
 	return b.autopilotConfig.DisableUpgradeMigration, true
 }
 
+// RunRaftWalVerifier runs a raft log store verifier in the background, if configured to do so.
+// This periodically writes out special raft logs to verify that the log store is not corrupting data.
+func (b *RaftBackend) RunRaftWalVerifier(ctx context.Context) {
+	if !b.verifierEnabled() {
+		return
+	}
+
+	go func() {
+		ticker := time.NewTicker(b.verificationInterval())
+		defer ticker.Stop()
+
+		logger := b.logger.Named("raft-wal-verifier")
+
+		for {
+			select {
+			case <-ticker.C:
+				err := b.applyVerifierCheckpoint()
+				if err != nil {
+					logger.Error("error applying verification checkpoint", "error", err)
+				}
+				logger.Debug("sent verification checkpoint")
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+}
+
+func (b *RaftBackend) applyVerifierCheckpoint() error {
+	data := make([]byte, 1)
+	data[0] = byte(verifierCheckpointOp)
+
+	b.permitPool.Acquire()
+	b.l.RLock()
+
+	var err error
+	applyFuture := b.raft.Apply(data, 0)
+	if e := applyFuture.Error(); e != nil {
+		err = e
+	}
+
+	// TODO: do we need to wait for the response?
+	_ = applyFuture.Response()
```

Comment:
No this is redundant as you are not using the Value anyway - in fact we already did the waiting up above when we called `applyFuture.Error` anyway.